### PR TITLE
Add --image= switch to firstboot, similar to --root= but with support for operating on disk image

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/core/machine-id-setup.c [hunk 1]
```diff
@@ -223,11 +223,9 @@ int machine_id_commit(const char *root) {
...
 
         /* Switch to a new mount namespace, isolate ourself and unmount etc_machine_id in our new namespace */
-        if (unshare(CLONE_NEWNS) < 0)
-                return log_error_errno(errno, "Failed to enter new namespace: %m");
-
-        if (mount(NULL, "/", NULL, MS_SLAVE | MS_REC, NULL) < 0)
-                return log_error_errno(errno, "Couldn't make-rslave / mountpoint in our private namespace: %m");
+        r = detach_mount_namespace();
+        if (r < 0)
+                return log_error_errno(errno, "Failed to set up new mount namespace: %m");
```
Comment (keszybz) [inline src/core/machine-id-setup.c]: errno → r

File: src/test/test-udev.c [hunk 2]
```diff
@@ -37,12 +38,11 @@ static int fake_filesystems(void) {
...
+        int r;
 
-        if (unshare(CLONE_NEWNS) < 0)
-                return log_error_errno(errno, "Failed to call unshare(): %m");
-
-        if (mount(NULL, "/", NULL, MS_SLAVE|MS_REC, NULL) < 0)
-                return log_error_errno(errno, "Failed to mount / as private: %m");
+        r = detach_mount_namespace();
+        if (r < 0)
+                return log_error_errno(errno, "Failed to detach mount namespace: %m");
```
Comment (keszybz) [inline src/test/test-udev.c]: errno → r

File: man/systemd-firstboot.xml [hunk 3]
```diff
@@ -259,6 +259,14 @@
         option should not be used lightly.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--welcome=</option></term>
+
+        <listitem><para>Takes a boolean argument. By default when prompting the user for configuration
+        options a brief welcome text is shown before the first question is asked. Use this switch to turn
+        off showing of the welcome text, by passing false.</para></listitem>
```
Comment (keszybz) [inline man/systemd-firstboot.xml]: "Pass false to this option to turn off the welcome text."

File: src/firstboot/firstboot.c [hunk 4]
```diff
@@ -940,6 +944,7 @@ static int help(void) {
                "     --setup-machine-id                     Generate a new random machine ID\n"
                "     --force                                Overwrite existing files\n"
                "     --delete-root-password                 Delete root password\n"
+               "     --welcome=no                           Disable showing of brief welcome text\n"
```
Comment (keszybz) [inline src/firstboot/firstboot.c]: "Disable the welcome text"

**Comment by poettering** [comment]
Force pushed a new version. No changes but the suggested ones.

**Comment by keszybz** [comment]
I don't know what happened with semaphoreci, but it doesn't look relevant.

**Pull request merged**